lib/sycamore/sycamore/connectors/file/file_scan.py

```diff
@@ -92,6 +99,28 @@ def _is_s3_scheme(self) -> bool:
         else:
             return all(path.startswith("s3:") for path in self._paths)
 
+    def _try_infer_fs(self):
```

Comment:
This seems related/overlapping with https://github.com/aryn-ai/sycamore/pull/925. Are they going to conflict?